### PR TITLE
v8 General Fixup

### PR DESCRIPTION
**Pull request by rekmarks**
Other fixups for the v8 docs excepting `docs/guide/rpc-api.md` (see #123) and `docs/guide/ethereum-provider.md` (see #134).

- `EIP XXXX` -> `EIP-XXXX` (per EIP style conventions)
- `web3` -> `web3.js` (for all references to `web3` the convenience library)
- `ethereum.enable()` -> `ethereum.request({ method: 'eth_requestAccounts' })`
- `ethereum.sendAsync` -> `ethereum.request`
- Delete `experimental-apis.md` (its content is consolidated in other pages via #123 and #134)
- Extensive editing/reformatting of `registering-function-names.md` (it was a hot and ugly mess)

### CODE REVIEW

**Comment by rekmarks** [comment]
@Gudahtt @whymarrh @BboyAkers @brad-decker 

I've made extensive updates to the whole repo. This is the miscellaneous/catchall PR. There are still significant improvements to be made elsewhere, but things brings the whole repo mostly up to date with the conventions and functionality of v8.

I've created a branch that's identical to `master`, https://github.com/MetaMask/metamask-docs/tree/v8, which I am using as a base to split up my work in the following PRs:
- #123 
- #134 
- #135 (this)

Once we've merged the above 3 PRs into `v8`, we can merge that into `master`. Then we can take further improvements from there.

If you want a view of the whole shebang, see #104. I'm probably just gonna close that one when we're done.

Cheers!

File: 00_Examples/00_Low_Level_Ether_Transfer.md [hunk 1]
```diff
@@ -60,42 +60,28 @@ function sendEtherFrom (account, callback) {
...
     from: account,
     to: yourAddress,
     value: value,
   }]
-  const from = account
-
-  // Now putting it all together into an RPC request:
-  const payload = {
-    method: method,
-    params: parameters,
```
Comment (Gudahtt) [inline 00_Examples/00_Low_Level_Ether_Transfer.md:73]: The old variable (`parameters`) is still referenced here

File: docs/.vuepress/components/SendTransaction.vue [hunk 2]
```diff
@@ -18,18 +18,19 @@ export default {
...
           gasPrice: 20000000000,
-        },
-        (result) => {
+        }]
+      })
+        .then((result) => {
           console.log(result);
-        }
-      );
+        });
```
Comment (Gudahtt) [inline docs/.vuepress/components/SendTransaction.vue:33]: Nit: We should probably not encourage people to ignore errors. Logging it is the least we could do.
```suggestion
        })
        .catch(console.error);
```

File: docs/guide/registering-function-names.md [hunk 3]
```diff
@@ -1,46 +1,45 @@
 # Registering Your Contract's Method Names
 
-MetaMask uses the parity on-chain registry of function signatures to display method names on the confirm screen. For many common method names, like token methods, this allows MetaMask to successfully look up the method names by their [method signature](https://solidity.readthedocs.io/en/v0.4.21/abi-spec.html). However, sometimes you're using a method that is not in that on-chain registry, and MetaMask will simply display `Unknown Function` to the user.
+MetaMask uses the Parity on-chain registry of function signatures to display method names on the confirm screen.
+For many common method names, like token methods, this allows MetaMask to successfully look up the method names by their [method signature](https://solidity.readthedocs.io/en/v0.4.21/abi-spec.html).
+However, sometimes you're using a method that is not in that on-chain registry, and MetaMask will simply display `Unknown Function` to the user.
```
Comment (Gudahtt) [inline docs/guide/registering-function-names.md:5]:
```suggestion
However, sometimes you're using a method that is not in that on-chain registry, and MetaMask will simply display `Contract Interaction` to the user.
```

This was a pre-existing mistake, but we show "Contract Interaction" in this case, not "Unknown Function".

Comment (Gudahtt) [reply docs/guide/registering-function-names.md:5]: Actually...  mobile might say something different in this scenario :thinking: Not sure what. I don't think it's "Unknown Function" though.

File: docs/guide/registering-your-token.md [hunk 4]
```diff
@@ -23,28 +23,31 @@ const tokenSymbol = 'TUT';
...
-    params: {
-      type: 'ERC20',
-      options: {
-        address: tokenAddress,
-        symbol: tokenSymbol,
-        decimals: tokenDecimals,
-        image: tokenImage,
+// wasAdded is a boolean. Like any RPC method, an error may be thrown.
+try {
+  const wasAdded = ethereum.request({
```
Comment (Gudahtt) [inline docs/guide/registering-your-token.md:28]:
```suggestion
  const wasAdded = await ethereum.request({
```

File: docs/guide/experimental-apis.md [hunk 5]
```diff
@@ -1,117 +0,0 @@
-# Experimental APIs
```
Comment (Gudahtt) [inline docs/guide/experimental-apis.md:1]: Was this included in this PR by mistake? It seems like it should have been in the other two.

File: 00_Examples/00_Low_Level_Ether_Transfer.md [hunk 6]
```diff
@@ -106,26 +92,31 @@ function pollForCompletion (txHash, callback) {
...
-    }, function (err, response) {
-      if (calledBack) return
-      if (err || response.error) {
+    })
+      .then((transaction) => {
+
+        if (calledBack) return
+
+        // We have successfully verified the mined transaction.
+        // Mind you, we should do this server side, with our own blockchain connection.
```
[the remark this answers is not in the document]
Comment (rekmarks) [reply 00_Examples/00_Low_Level_Ether_Transfer.md:104]: Truly perplexing stuff. Replacing with `// The transaction has been mined.`.

File: docs/guide/registering-your-token.md (hunk 4 above)
Comment (rekmarks) [reply docs/guide/registering-your-token.md]: ee70c19

File: docs/guide/experimental-apis.md (hunk 5 above)
Comment (rekmarks) [reply docs/guide/experimental-apis.md:1]: I decided to include it here, which I suppose is a bit kooky, but it should all make sense once the other two are merged in 😅

File: docs/guide/registering-function-names.md (hunk 3 above)
Comment (rekmarks) [reply docs/guide/registering-function-names.md]: I'll ask about mobile, but for now implemented your suggestion in ee70c19

File: docs/.vuepress/components/SendTransaction.vue (hunk 2 above)
Comment (rekmarks) [reply docs/.vuepress/components/SendTransaction.vue:33]: ee70c19

File: 00_Examples/00_Low_Level_Ether_Transfer.md [hunk 7]
```diff
@@ -106,26 +92,31 @@ function pollForCompletion (txHash, callback) {
...
+      })
+      .catch((error) => {
+
+        if (calledBack) return
+
         if (err.message.includes(notYet)) {
           return 'transaction is not yet mined'
         }
 
         callback(err || response.error)
```
[the remark this answers is not in the document]
Comment (rekmarks) [reply 00_Examples/00_Low_Level_Ether_Transfer.md]: This was actually even more outdated/broken, but I fixed it in ee70c19

File: 00_Examples/00_Low_Level_Ether_Transfer.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (rekmarks) [reply 00_Examples/00_Low_Level_Ether_Transfer.md]: ee70c19

**Comment by rekmarks** [comment]
Should be good to go with ee70c19!

**Pull request merged**